node/config/def.go

```diff
@@ -312,12 +315,13 @@ func DefaultStorageMiner() *StorageMiner {
 		Common: defCommon(),
 
 		Sealing: SealingConfig{
-			MaxWaitDealsSectors:       2, // 64G with 32G sectors
-			MaxSealingSectors:         0,
-			MaxSealingSectorsForDeals: 0,
-			WaitDealsDelay:            Duration(time.Hour * 6),
-			AlwaysKeepUnsealedCopy:    true,
-			FinalizeEarly:             false,
+			MaxWaitDealsSectors:        2, // 64G with 32G sectors
+			MaxSealingSectors:          0,
+			MaxSealingSectorsForDeals:  0,
+			WaitDealsDelay:             Duration(time.Hour * 6),
+			AlwaysKeepUnsealedCopy:     true,
+			FinalizeEarly:              false,
+			CollateralFromMinerBalance: false,
```

Comment:
Defaults to false as that won't work on fresh miners without sending them some funds first. We could add a separate option to prefer miner-available funds, and only send more if needed, but that requires a bit more advanced accounting logic (here we can rely on gas estimation failing in case of not having enough funds)

Reply:
Is the accounting any more advanced? Why is it not just a matter of setting `value = AvailableBalance - CollateralNeeded - Cfg.Buffer`, where `Buffer` is some minimum we don't want available balance to fall below.

Gas estimation should work exactly the same and catch any insufficient funds?